test/e2e/vsphere_volume_placement.go

```diff
@@ -19,218 +19,412 @@ package e2e
-			vsp.DeleteVolume(volumePath)
-		}
 	})
 
 	framework.KubeDescribe("provision pod on node with matching labels", func() {
+		var (
+			vsp         *vsphere.VSphere
+			volumePaths []string
+			err         error
+		)
+		/*
+			Steps
+			1. Create VMDK volume
+			2. Find two nodes with the status available and ready for scheduling.
+			3. Add labels to the both nodes. - (vsphere_e2e_label: Random UUID)
+
+		*/
+		BeforeEach(func() {
+			if !testSeup {
+				c = f.ClientSet
+				ns = f.Namespace.Name
+				framework.ExpectNoError(framework.WaitForAllNodesSchedulable(c, framework.TestContext.NodeSchedulableTimeout))
+				nodeList := framework.GetReadySchedulableNodesOrDie(f.ClientSet)
+				node1Name, node1KeyValueLabel, node2Name, node2KeyValueLabel = testSetupVolumePlacement(c, ns, nodeList)
+				testSeup = true
+			}
+			By("creating vmdk")
+			vsp, err = vsphere.GetVSphere()
+			Expect(err).NotTo(HaveOccurred())
+			volumePath, err := createVSphereVolume(vsp, nil)
+			Expect(err).NotTo(HaveOccurred())
+			volumePaths = append(volumePaths, volumePath)
+		})
 
+		AfterEach(func() {
+			if len(volumePaths) > 0 {
+				vsp, err := vsphere.GetVSphere()
+				Expect(err).NotTo(HaveOccurred())
+				for _, volumePath := range volumePaths {
+					vsp.DeleteVolume(volumePath)
+				}
+			}
+			volumePaths = nil
+		})
 		/*
 			Steps
 
-			1. Create POD Spec with volume path of the vmdk and NodeSelector set to label assigned to node1.
-			2. Create POD and wait for POD to become ready.
+			1. Create pod Spec with volume path of the vmdk and NodeSelector set to label assigned to node1.
+			2. Create pod and wait for pod to become ready.
 			3. Verify volume is attached to the node1.
-			4. Delete POD.
-			5. Wait for volume to be detached from the node1.
-			6. Repeat Step 1 to 5 and make sure back to back pod creation on same worker node with the same volume is working as expected.
+			4. Create empty file on the volume to verify volume is writable.
+			5. Verify newly created file and previously created files exist on the volume.
+			6. Delete pod.
+			7. Wait for volume to be detached from the node1.
+			8. Repeat Step 1 to 7 and make sure back to back pod creation on same worker node with the same volume is working as expected.
 
 		*/
 
 		It("should create and delete pod with the same volume source on the same worker node", func() {
-			pod := createPodWithVolumeAndNodeSelector(c, ns, vsp, node1Name, node1KeyValueLabel, volumePath)
-			deletePodAndWaitForVolumeToDetach(c, ns, vsp, node1Name, pod, volumePath)
+			var volumeFiles []string
+			pod := createPodWithVolumeAndNodeSelector(c, ns, vsp, node1Name, node1KeyValueLabel, volumePaths)
+
+			// Create empty files on the mounted volumes on the pod to verify volume is writable
+			// Verify newly and previously created files present on the volume mounted on the pod
+			newEmptyFileName := "/mnt/volume1/" + ns + "_1" + ".txt"
+			volumeFiles = append(volumeFiles, newEmptyFileName)
```

Comment:
fixed. need a slice to pass to createAndVerifyFilesOnVolume, some tests requires to create multiple files on multiple volumes.